### ISSUE TITLE
localization code generation doesn't work

### ISSUE THREAD

**Issue by evanholt1** [flutter/flutter#76980]
Hi!
I am following the tutorial for Internationalizing flutter Apps,
https://flutter.dev/docs/development/accessibility-and-localization/internationalization
and I have reached step 6 where you run your app and code generation happens, 
yet that does not happen.
I run the application, yet no code is generated. and subsequent steps which need 
`import 'package:flutter_gen/gen_l10n/app_localizations.dart'; `
don't work as flutter_gen is not recognized.

I have made sure the I10n file is not in the lib folder. I have manually created the .dart-tools/flutter_gen/pubspec.yaml file and a name key, yet code is not generated. nothing works.

 
## Steps to Reproduce
- create a new flutter project
- follow tutorial till step 6
- expect code generation, but it doesn't occur
 

**Expected results:** 
- I wanted the expected code generation file. the flutter_gen file in .dart-tools is not created and no code is generated. subsequently, importing flutter_gen in the next steps doesn't work.

**Actual results:** 
- no code generation occurs. no flutter_gen folder is created. importing flutter_gen in the main file doesn't work

**main.dart**
```
import 'package:flutter/material.dart';
import 'package:flutter_localizations/flutter_localizations.dart';
// TODO: uncomment the line below after codegen
// import 'package:flutter_gen/gen_l10n/app_localizations.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Text("here"),
      localizationsDelegates: [
        // ... app-specific localization delegate[s] here
        // TODO: uncomment the line below after codegen
        // AppLocalizations.delegate,
        GlobalMaterialLocalizations.delegate,
        GlobalWidgetsLocalizations.delegate,
        GlobalCupertinoLocalizations.delegate,
      ],
      supportedLocales: [
        const Locale('en', ''), // English, no country code
        const Locale('es', ''), // Spanish, no country code
      ],
    );
  }
}
```

**pubspec.yaml**
```
name: flutter_app
description: A new Flutter application.

publish_to:

version: 1.0.0+1

environment:
  sdk: ">=2.7.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter
  flutter_localizations:
    sdk: flutter
  intl: ^0.16.1
  cupertino_icons: ^1.0.0

dev_dependencies:
  flutter_test:
    sdk: flutter

flutter:
  uses-material-design: true
  generate: true
```

**I10n.yaml**
```
arb-dir: lib/l10n
template-arb-file: app_en.arb
output-localization-file: app_localizations.dart
```

**lib/I10n/app_en.arb**
```
{
  "helloWorld": "Hello World!"
}
```

<details> 
<summary>flutter doctor -v</summary> 

```bash
<!-- Finally, paste the output of running `flutter doctor -v` here. -->
[✓] Flutter (Channel stable, 1.22.6, on macOS 11.2.2 20D80 darwin-x64, locale en-JO)
    • Flutter version 1.22.6 at /Users/mac/development/flutter
    • Framework revision 9b2d32b605 (5 weeks ago), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/mac/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✗] Xcode - develop for iOS and macOS
    ✗ Xcode installation is incomplete; a full installation is necessary for iOS development.
      Download at: https://developer.apple.com/xcode/download/
      Or install Xcode via the App Store.
      Once installed, run:
        sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
        sudo xcodebuild -runFirstLaunch
    ✗ CocoaPods not installed.
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To install:
        sudo gem install cocoapods

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] Connected device (1 available)
    • sdk gphone x86 (mobile) • emulator-5554 • android-x86 • Android 11 (API 30) (emulator)

```

</details>

I am using android studio, thus the flutter and dart plugins aren't registered, yet everything run fine as they are installed.

**Comment by TahaTesser** [comment]
Hi @evanholt1 
Thanks for filing the issue, I experienced this recently, this is most likely due to `flutter_localizations` on the stable

Since `intl 0.17.0` requires 2.12, we can't use `flutter_localizations` with `stable`

```bash
[  +12 ms] SLVR:             conflict: every version of flutter_localizations from sdk depends on intl 0.17.0
[        ] SLVR:             ! flutter_localizations any from sdk is satisfied by flutter_localizations any from sdk
[        ] SLVR:             ! which is caused by "stable_flutter depends on flutter_localizations any from sdk"
[        ] SLVR:             ! thus: intl 0.17.0 is required
```



<details>
<summary>logs</summary>

```bash
<0.13.0"}},"archive_url":"https://pub.dartlang.org/packages/string_scanner/versions/0.1.3%2B2.tar.gz","published":"2015-0
9-02T00:29:50.320Z"},{"version":"0.1.4","pubspec":{"version":"0.1.4","name":"string_scanner","dependencies":{"charcode":"
^1.1.0","source_span":"^1.0.0","path":"^1.2.0"},"author":"Dart Team <misc@dartlang.org>","description":"A class for
parsing strings using a sequence of patterns.\n","environment":{"sdk":">=1.8.0
<2.0.0"},"homepage":"https://github.com/dart-lang/string_scanner","dev_dependencies":{"test":">=0.12.0
<0.13.0"}},"archive_url":"https://pub.dartlang.org/packages/string_scanner/versions/0.1.4.tar.gz","published":"2015-09-02
T18:45:57.724Z"},{"version":"0.1.4+1","pubspec":{"version":"0.1.4+1","name":"string_scanner","dependencies":{"charcode":"
^1.1.0","source_span":"^1.0.0"},"author":"Dart Team <misc@dartlang.org>","description":"A class for parsing strings using
a sequence of patterns.\n","environment":{"sdk":">=1.8.0
<2.0.0"},"homepage":"https://github.com/dart-lang/string_scanner","dev_dependencies":{"test":">=0.12.0
<0.13.0"}},"archive_url":"https://pub.dartlang.org/packages/string_scanner/versions/0.1.4%2B1.tar.gz","published":"2016-0
1-15T00:34:16.276Z"},{"version":"0.1.5","pubspec":{"version":"0.1.5","name":"string_scanner","dependencies":{"charcode":"
^1.1.0","source_span":"^1.0.0"},"author":"Dart Team <misc@dartlang.org>","description":"A class for parsing strings using
a sequence of patterns.\n","environment":{"sdk":">=1.8.0
<2.0.0"},"homepage":"https://github.com/dart-lang/string_scanner","dev_dependencies":{"test":">=0.12.0
<0.13.0"}},"archive_url":"https://pub.dartlang.org/packages/string_scanner/versions/0.1.5.tar.gz","published":"2016-06-06
T23:52:28.189Z"},{"version":"1.0.0","pubspec":{"version":"1.0.0","name":"string_scanner","dependencies":{"charcode":"^1.1
.0","source_span":"^1.0.0"},"author":"Dart Team <misc@dartlang.org>","description":"A class for parsing strings using a
sequence of patterns.\n","environment":{"sdk":">=1.8.0
<2.0.0"},"homepage":"https://github.com/dart-lang/string_scanner","dev_dependencies":{"test":">=0.12.0
<0.13.0"}},"archive_url":"https://pub.dartlang.org/packages/string_scanner/versions/1.0.0.tar.gz","published":"2016-06-14
T23:21:48.009Z"},{"version":"1.0.1","pubspec":{"version":"1.0.1","name":"string_scanner","dependencies":{"charcode":"^1.1
.0","source_span":"^1.0.0"},"author":"Dart Team <misc@dartlang.org>","description":"A class for parsing strings using a
sequence of patterns.\n","environment":{"sdk":">=1.8.0
<2.0.0"},"homepage":"https://github.com/dart-lang/string_scanner","dev_dependencies":{"test":">=0.12.0
<0.13.0"}},"archive_url":"https://pub.dartlang.org/packages/string_scanner/versions/1.0.1.tar.gz","published":"2016-11-02
T22:12:48.016Z"},{"version":"1.0.2","pubspec":{"version":"1.0.2","name":"string_scanner","dependencies":{"charcode":"^1.1
.0","source_span":"^1.4.0"},"author":"Dart Team <misc@dartlang.org>","description":"A class for parsing strings using a
sequence of patterns.\n","environment":{"sdk":">=1.8.0
<2.0.0"},"homepage":"https://github.com/dart-lang/string_scanner","dev_dependencies":{"test":">=0.12.0
<0.13.0"}},"archive_url":"https://pub.dartlang.org/packages/string_scanner/versions/1.0.2.tar.gz","published":"2017-05-22
T19:51:33.942816Z"},{"version":"1.0.3","pubspec":{"version":"1.0.3","name":"string_scanner","dependencies":{"charcode":"^
1.1.0","source_span":"^1.4.0"},"author":"Dart Team <misc@dartlang.org>","description":"A class for parsing strings using
a sequence of patterns.","homepage":"https://github.com/dart-lang/string_scanner","environment":{"sdk":">=1.8.0
<3.0.0"},"dev_dependencies":{"test":">=0.12.0
<2.0.0"}},"archive_url":"https://pub.dartlang.org/packages/string_scanner/versions/1.0.3.tar.gz","published":"2018-07-18T
19:31:40.796230Z"},{"version":"1.0.4","pubspec":{"version":"1.0.4","name":"string_scanner","dependencies":{"charcode":"^1
.1.0","source_span":"^1.4.0","meta":"^1.1.0"},"author":"Dart Team <misc@dartlang.org>","description":"A class for parsing
strings using a sequence of
patterns.","homepage":"https://github.com/dart-lang/string_scanner","environment":{"sdk":">=1.8.0
<3.0.0"},"dev_dependencies":{"test":">=0.12.0
<2.0.0"}},"archive_url":"https://pub.dartlang.org/packages/string_scanner/versions/1.0.4.tar.gz","published":"2018-08-13T
19:34:54.805889Z"},{"version":"1.0.5","pubspec":{"version":"1.0.5","name":"string_scanner","dependencies":{"charcode":"^1
.1.0","source_span":"^1.4.0","meta":"^1.1.0"},"author":"Dart Team <misc@dartlang.org>","description":"A class for parsing
strings using a sequence of
patterns.","homepage":"https://github.com/dart-lang/string_scanner","environment":{"sdk":">=2.0.0
<3.0.0"},"dev_dependencies":{"test":"^1.0.0"}},"archive_url":"https://pub.dartlang.org/packages/string_scanner/versions/1
.0.5.tar.gz","published":"2019-07-31T20:18:04.223091Z"},{"version":"1.1.0-nullsafety","pubspec":{"name":"string_scanner",
"version":"1.1.0-nullsafety","description":"A class for parsing strings using a sequence of
patterns.","homepage":"https://github.com/dart-lang/string_scanner","environment":{"sdk":">=2.10.0-0
<2.10.0"},"dependencies":{"charcode":">=1.2.0-nullsafety <1.2.0","source_span":">=1.8.0-nullsafety
<1.8.0"},"dev_dependencies":{"test":">=1.16.0-nullsafety
<1.16.0"}},"archive_url":"https://pub.dartlang.org/packages/string_scanner/versions/1.1.0-nullsafety.tar.gz","published":
"2020-07-29T16:01:36.464304Z"},{"version":"1.1.0-nullsafety.1","pubspec":{"name":"string_scanner","version":"1.1.0-nullsa
fety.1","description":"A class for parsing strings using a sequence of
patterns.","homepage":"https://github.com/dart-lang/string_scanner","environment":{"sdk":">=2.10.0-0
<2.11.0"},"dependencies":{"charcode":">=1.2.0-nullsafety <1.2.0","source_span":">=1.8.0-nullsafety
<1.8.0"},"dev_dependencies":{"test":">=1.16.0-nullsafety
<1.16.0"}},"archive_url":"https://pub.dartlang.org/packages/string_scanner/versions/1.1.0-nullsafety.1.tar.gz","published
":"2020-09-22T16:20:09.152273Z"},{"version":"1.1.0-nullsafety.2","pubspec":{"name":"string_scanner","version":"1.1.0-null
safety.2","description":"A class for parsing strings using a sequence of
patterns.","homepage":"https://github.com/dart-lang/string_scanner","environment":{"sdk":">=2.10.0-0
<2.12.0"},"dependencies":{"charcode":">=1.2.0-nullsafety <1.2.0","source_span":">=1.8.0-nullsafety
<1.8.0"},"dev_dependencies":{"test":">=1.16.0-nullsafety
<1.16.0"}},"archive_url":"https://pub.dartlang.org/packages/string_scanner/versions/1.1.0-nullsafety.2.tar.gz","published
":"2020-10-23T21:32:54.442267Z"},{"version":"1.1.0-nullsafety.3","pubspec":{"name":"string_scanner","version":"1.1.0-null
safety.3","description":"A class for parsing strings using a sequence of
patterns.","homepage":"https://github.com/dart-lang/string_scanner","environment":{"sdk":">=2.12.0-0
<3.0.0"},"dependencies":{"charcode":">=1.2.0-nullsafety <1.2.0","source_span":">=1.8.0-nullsafety
<1.8.0"},"dev_dependencies":{"test":">=1.16.0-nullsafety
<1.16.0"}},"archive_url":"https://pub.dartlang.org/packages/string_scanner/versions/1.1.0-nullsafety.3.tar.gz","published
":"2020-11-03T22:35:52.697298Z"},{"version":"1.1.0","pubspec":{"name":"string_scanner","version":"1.1.0","description":"A
class for parsing strings using a sequence of
patterns.","homepage":"https://github.com/dart-lang/string_scanner","environment":{"sdk":">=2.12.0-0
<3.0.0"},"dependencies":{"charcode":"^1.2.0","source_span":"^1.8.0"},"dev_dependencies":{"test":"^1.16.0-nullsafety"}},"a
rchive_url":"https://pub.dartlang.org/packages/string_scanner/versions/1.1.0.tar.gz","published":"2021-02-02T23:50:11.994
548Z"}],"_fetchedAt":"2021-03-01T19:21:47.933781"}
[  +21 ms] SLVR:     fact: shared_preferences 0.5.12+4 depends on meta ^1.0.4
[        ] SLVR:     fact: shared_preferences 0.5.12+4 depends on flutter any from sdk
[        ] SLVR:     fact: shared_preferences 0.5.12+4 depends on shared_preferences_platform_interface ^1.0.0
[        ] SLVR:     fact: shared_preferences 0.5.12+4 depends on shared_preferences_linux ^0.0.2
[        ] SLVR:     fact: shared_preferences 0.5.12+4 depends on shared_preferences_macos ^0.0.1
[        ] SLVR:     fact: shared_preferences 0.5.12+4 depends on shared_preferences_web ^0.1.2
[        ] SLVR:     fact: shared_preferences 0.5.12+4 depends on shared_preferences_windows ^0.0.1
[        ] SLVR:       selecting shared_preferences 0.5.12+4
[        ] SLVR:       derived: shared_preferences_windows ^0.0.1
[        ] SLVR:       derived: shared_preferences_web ^0.1.2
[   +2 ms] SLVR:       derived: shared_preferences_macos ^0.0.1
[        ] SLVR:       derived: shared_preferences_linux ^0.0.2
[        ] SLVR:       derived: shared_preferences_platform_interface ^1.0.0
[        ] SLVR:       fact: path_provider 1.6.27 depends on flutter any from sdk
[        ] SLVR:       fact: path_provider 1.6.27 depends on path_provider_platform_interface ^1.0.1
[        ] SLVR:       fact: path_provider 1.6.27 depends on path_provider_macos ^0.0.4
[        ] SLVR:       fact: path_provider 1.6.27 depends on path_provider_linux ^0.0.1
[        ] SLVR:       fact: path_provider 1.6.27 depends on path_provider_windows ^0.0.4
[        ] SLVR:         selecting path_provider 1.6.27
[        ] SLVR:         derived: path_provider_windows ^0.0.4
[        ] SLVR:         derived: path_provider_linux ^0.0.1
[  +13 ms] SLVR:         derived: path_provider_macos ^0.0.4
[        ] SLVR:         derived: path_provider_platform_interface ^1.0.1
[        ] SLVR:           selecting cupertino_icons 1.0.0
[        ] SLVR:           fact: intl 0.16.1 depends on path >=0.9.0 <2.0.0
[        ] SLVR:             selecting intl 0.16.1
[        ] SLVR:             fact: flutter_localizations 0.0.0 from sdk depends on flutter any from sdk
[        ] SLVR:             fact: flutter_localizations 0.0.0 from sdk depends on intl 0.17.0
[        ] SLVR:             fact: flutter_localizations 0.0.0 from sdk depends on characters 1.1.0
[        ] SLVR:             fact: flutter_localizations 0.0.0 from sdk depends on clock 1.1.0
[        ] SLVR:             fact: flutter_localizations 0.0.0 from sdk depends on collection 1.15.0
[        ] SLVR:             fact: flutter_localizations 0.0.0 from sdk depends on meta 1.3.0
[        ] SLVR:             fact: flutter_localizations 0.0.0 from sdk depends on path 1.8.0
[        ] SLVR:             fact: flutter_localizations 0.0.0 from sdk depends on typed_data 1.3.0
[  +12 ms] SLVR:             fact: flutter_localizations 0.0.0 from sdk depends on vector_math 2.1.0
[        ] SLVR:             derived: not flutter_localizations 0.0.0 from sdk
[        ] SLVR:             fact: every version of flutter_localizations from sdk depends on characters 1.1.0
[        ] SLVR:             fact: every version of flutter_localizations from sdk depends on clock 1.1.0
[        ] SLVR:             fact: every version of flutter_localizations from sdk depends on collection 1.15.0
[        ] SLVR:             fact: every version of flutter_localizations from sdk depends on flutter any from sdk
[        ] SLVR:             fact: every version of flutter_localizations from sdk depends on intl 0.17.0
[        ] SLVR:             fact: every version of flutter_localizations from sdk depends on meta 1.3.0
[        ] SLVR:             fact: every version of flutter_localizations from sdk depends on path 1.8.0
[        ] SLVR:             fact: every version of flutter_localizations from sdk depends on typed_data 1.3.0
[        ] SLVR:             fact: every version of flutter_localizations from sdk depends on vector_math 2.1.0
[  +12 ms] SLVR:             conflict: every version of flutter_localizations from sdk depends on intl 0.17.0
[        ] SLVR:             ! flutter_localizations any from sdk is satisfied by flutter_localizations any from sdk
[        ] SLVR:             ! which is caused by "stable_flutter depends on flutter_localizations any from sdk"
[        ] SLVR:             ! thus: intl 0.17.0 is required
[        ] SLVR:             ! not intl 0.17.0 is satisfied by intl 0.16.1
[        ] SLVR:             ! which is caused by "stable_flutter depends on intl 0.16.1"
[        ] SLVR:             ! thus: version solving failed
[        ] SLVR: Version solving took 0:00:01.555442 seconds.
[        ]     | Tried 1 solutions.
[        ] FINE: Resolving dependencies finished (1.606s).
[        ] ERR : Because stable_flutter depends on flutter_localizations any from sdk which depends on intl 0.17.0, intl
0.17.0 is required.
[        ]     | So, because stable_flutter depends on intl 0.16.1, version solving failed.
[        ] FINE: Exception type: SolveFailure
[        ] FINE: package:pub/src/solver/version_solver.dart 308:5
VersionSolver._resolveConflict
[        ]     | package:pub/src/solver/version_solver.dart 129:27
VersionSolver._propagate
[        ]     | package:pub/src/solver/version_solver.dart 93:11
VersionSolver.solve.<fn>
[        ]     | ===== asynchronous gap ===========================
[        ]     | dart:async                                                                            Future.catchError
[        ]     | package:pub/src/utils.dart 113:52
captureErrors.wrappedCallback
[        ]     | package:stack_trace                                                                   Chain.capture
[        ]     | package:pub/src/utils.dart 126:11                                                     captureErrors
[        ]     | package:pub/src/command.dart 164:13                                                   PubCommand.run
[        ]     | package:args/command_runner.dart 196:27
CommandRunner.runCommand
[        ]     | package:pub/src/command_runner.dart 150:26
PubCommandRunner.runCommand
[        ]     | package:pub/src/command_runner.dart 138:18
PubCommandRunner.run
[        ]     | /opt/s/w/ir/cache/builder/src/third_party/dart/third_party/pkg/pub/bin/pub.dart 9:48  main
[  +39 ms] Running "flutter pub get" in stable_flutter... (completed in 2,126ms)
[        ] "flutter run" took 9,349ms.
[   +4 ms] pub get failed (1;     | /opt/s/w/ir/cache/builder/src/third_party/dart/third_party/pkg/pub/bin/pub.dart 9:48
main)
[        ] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      _DefaultPub.batch (package:flutter_tools/src/dart/pub.dart:322:7)
           <asynchronous suspension>
           #2      _DefaultPub.get (package:flutter_tools/src/dart/pub.dart:222:7)
           <asynchronous suspension>
           #3      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:1135:7)
           <asynchronous suspension>
           #4      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1009:27)
           <asynchronous suspension>
           #5      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #6      AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #7      CommandRunner.runCommand (package:args/command_runner.dart:197:13)
           <asynchronous suspension>
           #8      FlutterCommandRunner.runCommand.<anonymous closure>
           (package:flutter_tools/src/runner/flutter_command_runner.dart:278:9)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #10     AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #11     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:234:5)
           <asynchronous suspension>
           #12     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:64:9)
           <asynchronous suspension>
           #13     run.<anonymous closure> (package:flutter_tools/runner.dart:62:12)
           <asynchronous suspension>
           #14     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #15     AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #16     runInContext (package:flutter_tools/src/context_runner.dart:73:10)
           <asynchronous suspension>
           #17     main (package:flutter_tools/executable.dart:90:3)
           <asynchronous suspension>
           
           
[ +256 ms] ensureAnalyticsSent: 254ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1
tahatesser@Tahas-MacBook-Pro stable_flutter % 
```

</details>


### Workaround
Upgrade to the beta channel 

I omitted `intl`, tried running the app, and surprisingly code gen is still done and the app launches

<img width="936" alt="Screenshot 2021-03-01 at 7 48 41 PM" src="https://user-images.githubusercontent.com/48603081/109509786-4767d600-7ac7-11eb-845a-cda5967e489c.png">


<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel stable, 1.22.6, on macOS 11.2.2 20D80 darwin-x64, locale en-GB)
    • Flutter version 1.22.6 at /Users/tahatesser/Code/flutter_stable
    • Framework revision 9b2d32b605 (5 weeks ago), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.4)
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.53.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.19.0

[✓] Connected device (2 available)
    • SM M025F (mobile)    • R9ZR205XX0A               • android-arm • Android 10 (API 29)
    • Taha’s iPad (mobile) • 00008020-000255113EE8402E • ios         • iOS 14.4

! Doctor found issues in 1 category.
```

</details>

**Comment by evanholt1** [comment]
hello! sorry for bothering you1
so I started a new flutter project in android studio, 
changed to the beta channel,
then updated to the newest beta version,
then updated the intl package to 0.17.0,
then commented the flutter_gen code and tried running the code again.
and... it didn't work. still no code generation. 
I manually even created .dart-tools/flutter_gen/gen_I10n/app_localizations.dart (file & directories),
yet nothing.
I also tried removing the intl package but also didn't work.\
again, this is a new empty flutter project.
also I tried running with a terminal ,and with android studio's run button.

<img width="2048" alt="Screen Shot 2021-03-01 at 4 39 06 PM" src="https://user-images.githubusercontent.com/40251685/109512968-4fb31780-7aad-11eb-8a07-4d672006c6be.png">

- pubspec.yaml
```
name: gen_l10n_example
description: Example of an internationalized Flutter app that uses `flutter gen-l10n`

environment:
  sdk: '>=2.10.0 <3.0.0'

dependencies:
  flutter:
    sdk: flutter
  # Internationalization support.
  flutter_localizations:
    sdk: flutter


flutter:
  generate: true # Add this line
  uses-material-design: true
```

**Comment by TahaTesser** [comment]
Hi @evanholt1 
You have commented the import, please uncomment (do not create `dart-tools/flutter_gen/gen_I10n/app_localizations.dart` manually), when you uncomment `flutter_gen` import, it will show an error but ignore and execute `flutter run` from the terminal, this will generate the files and error from import will go away.

**Comment by evanholt1** [comment]
sooooo..... It worked!
and the solution was stupidly simple. it was a mis-labelling. 
I was using I10n( with a capital letter I) instead of l10n( with a lowercase l). it occurred when I learned l10n stands for Localisation. 

so I changed the names of I10n.yaml and the I10n folder from capital 'i's to lowercase 'L' s, and the codegen worked then. I feel so stupid xD 

I would like to point out as well, that I didn't even need now to make imports anywhere, just adding the dependence to pubspec.yaml and and the 'generate: true' flag, and defining the l10n.yaml & l10n folder, was enough for the codegen to work properly.

thanks for the help. I just wish there is a note to help with this issue, as everywhere nowadays, capital I is written the exact same as lowercase L.
